src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -493,21 +493,27 @@ internal static string TupleMemberName(int position)
             return "Item" + position;
         }
 
-        // This array needs to be sorted for binary searching
-        private static readonly string[] reservedMemberNames = new[]
-        {
-            "CombineHashCodes",
-            "CompareTo",
-            "Create",
-            "Deconstruct",
-            "Equals",
-            "GetHashCode",
-            "GetHashCodeCore",
-            "Rest",
-            "Size",
-            "ToString",
-            "ToStringEnd"
-        };
+        private static bool IsElementNameForbidden(string name)
+        {
+            switch (name)
+            {
+                case "CombineHashCodes":
+                case "CompareTo":
+                case "Create":
+                case "Deconstruct":
+                case "Equals":
+                case "GetHashCode":
+                case "GetHashCodeCore":
+                case "Rest":
```

Comment:
Correct. If you wrote `var x = (First: 1, Rest: 2); M(x.Rest);` it would be ambiguous which `Rest` you meant.
The change above just catches that early.